File: GroceryListHelper.Client/Interfaces/ICartProductsService.cs
namespace GroceryListHelper.Client.Interfaces;

public interface ICartProductsService
{
    Task<List<CartProductCollectable>> GetCartProducts();
    Task DeleteAllCartProducts();
    Task DeleteCartProduct(string name);
    Task CreateCartProduct(CartProduct product);
    Task UpdateCartProduct(CartProductCollectable cartProduct);
    Task SortCartProducts(ListSortDirection sortDirection);
}

File: GroceryListHelper.Client/src/types/ICartProductsService.ts
import type { CartProduct } from "./CartProduct";
import type { SortDirection } from "./SortState";

export interface ICartProductsService {
    getCartProducts: () => Promise<CartProduct[] | Error>;
    createCartProduct: (product: CartProduct) => Promise<Error | null>;
    updateCartProduct: (product: CartProduct) => Promise<Error | null>;
    sortCartProducts: (sortDirection: SortDirection) => Promise<Error | null>;
    deleteCartProduct: (name: string) => Promise<Error | null>;
    deleteAllCartProducts: () => Promise<Error | null>;
}